torch/csrc/jit/codegen/cuda/lower_expr_sort.cpp

```diff
@@ -691,17 +691,21 @@ struct LocalDomainSorter {
         concrete_id_dependencies_.end()) {
       const auto& dependencies_0 = concrete_id_dependencies_.at(concrete_id_0);
       // if id0 depends on id1 it means id1 is inside id0, so id0 < id1
-      return dependencies_0.count(concrete_id_1);
+      if (dependencies_0.count(concrete_id_1)) {
+        return true;
+      }
     }
 
     if (concrete_id_dependencies_.find(concrete_id_1) !=
         concrete_id_dependencies_.end()) {
       const auto& dependencies_1 = concrete_id_dependencies_.at(concrete_id_1);
       // if id1 depends on id0 it means id0 is inside id1, so id1 < id0
-      return !dependencies_1.count(concrete_id_0);
+      if (dependencies_1.count(concrete_id_0)) {
```

Comment:
This doesn't look right... We can only return `false` after this point. In the old code, we could return `true` if `dependencies_1.count(concrete_id_0) == false`

Reply:
this is my understanding, but lmk if something seems wrong here:

- this looks like it's basically trying to do a toposort, where, if a depends on b, then we want a to go before b in the final sorted order. afaik, the typical toposort comparator is "if a path a->b exists, then comp(a, b) = true. Otherwise comp(a, b) = false". Or in this case, if a maps to a set containing b, then return comp(a, b) = true. In particular, if there's no relationship between a and b, then comp(a, b) and comp(b, a) should both be false (unless you have some other way to break ties in this case)

if this is a toposort, then `dependencies_1.count(concrete_id_0) == false` is a case where  we don't necessarily have any dependence between `dep_1` and `dep_0`; in other words, `0 -> 1` (dep_0 maps to dep_1) implies we should return true; and `0 -> 1` implies `1 -/-> 0`; but `1 -/-> 0` doesn't imply `0 -> 1`, so `1 -/-> 0` doesn't necessarily mean we should return true.

Reply:
This sounds very reasonable to me (without knowing much about the actual implementation).

I'd argue in that instance, the entire block inside could just be removed, since it only returns `false` inside and that is the default behavior after the true block. https://github.com/pytorch/pytorch/blob/9b49a26918f1a91daf34233036eddb90d1a2ae71/torch/csrc/jit/codegen/cuda/lower_expr_sort.cpp#L699-L706

Having said that, I'm waiting on someone who actually knows what this does to respond. cc'ing @csarofeen @naoyam

Reply:
yeah, that's true - I can delete those blocks if we decide this change is okay

Reply:
Thanks @davidberard98. I believe the fix makes sense and is indeed a nice catch.

Reply:
great, thanks for the help reviewing @naoyam @jjsjann123 

I removed the false blocks like @jjsjann123 suggested